crates/burn-wgpu/src/compiler/wgsl/base.rs

```diff
@@ -218,7 +218,9 @@ impl Display for Variable {
             }
             Variable::ConstantScalar(number, elem) => match elem {
                 Elem::F32 => f.write_fmt(format_args!("{number}f")),
-                Elem::I32 => f.write_fmt(format_args!("{number}i")),
+                // Dawn seems to get tripped up by the 'i' suffix, while wgpu is happy
+                // with or without it, so emit the literal without it.
+                Elem::I32 => f.write_fmt(format_args!("{number}")),
```

Comment:
sanity question: wgpu is still correctly interpreting this as a signed int, right?

Reply:
I've not seen any evidence to the contrary. The standard says that literals without a suffix should be treated as `AbstractInt` (https://www.w3.org/TR/WGSL/#numeric-literal) with values in the range `[-2^63, 2^63)` (https://www.w3.org/TR/WGSL/#abstractint). So, I think the answer is yes :)